packages/gestalt/src/contexts/Theme.js

```diff
@@ -0,0 +1,132 @@
+// @flow strict
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+export type ColorScheme = 'light' | 'dark' | 'userPreference';
+
+export const ColorSchemePropType: React$PropType$Primitive<ColorScheme> = PropTypes.oneOf(
+  ['light', 'dark', 'userPreference']
+);
+
+type Theme = {|
+  name: string,
+  colorRed0: string,
+  colorRed100: string,
+  colorGray0: string,
+  colorGray50: string,
+  colorGray100: string,
+  colorGray200: string,
+  colorGray300: string,
+  colorGray400: string,
+|};
+
+type Props = {|
+  children: React.Node,
+  colorScheme?: ColorScheme,
+|};
+
+const lightModeTheme = {
+  name: 'lightMode',
+  colorRed0: '#ff5247',
+  colorRed100: '#e60023',
+  colorGray0: '#fff',
+  colorGray50: '#fff',
+  colorGray100: '#efefef',
+  colorGray200: '#767676',
+  colorGray300: '#111',
+  colorGray400: '#000',
+};
+
+const darkModeTheme = {
+  name: 'darkMode',
+  colorRed0: '#e60023',
+  colorRed100: '#ff5247',
+  colorGray0: '#050505',
+  colorGray50: '#272727',
+  colorGray100: '#494949',
+  colorGray200: '#b8b8b8',
+  colorGray300: '#efefef',
+  colorGray400: '#fff',
+};
+
+const ThemeContext: React.Context<Theme> = React.createContext<Theme>(
+  lightModeTheme
+);
+
+const themeToStyles = theme => {
+  let styles = '';
+  Object.keys(theme).forEach(key => {
+    if (key.startsWith('color')) {
+      styles += `  --${key}: ${theme[key]};\n`;
+    }
+  });
+  return styles;
+};
+
+let themeId = 0;
+const getThemeId = () => {
+  themeId += 1;
+  return themeId;
+};
+
+const getTheme = (colorScheme: ?ColorScheme) =>
+  colorScheme === 'dark' ||
+  (colorScheme === 'userPreference' &&
+    typeof window !== 'undefined' &&
+    window.matchMedia &&
+    window.matchMedia('(prefers-color-scheme: dark)').matches)
+    ? darkModeTheme
+    : lightModeTheme;
+
+export function ThemeProvider({
```

Comment:
The naming here could be a bit more consistent with the CSS Spec: `<ColorSchemeProvider prefers="dark" />`

Reply:
We've had enough requests for theming that I think we'll need to provide something more flexible than locking it down to only light/dark in the future. I'd prefer to keep the name more generically "theme" for now so it'll still make sense then. Also, the data in the context here is an object containing colors so it feels more "theme-y" than just being a binary color preference even though that's how the specific theme got set.